### PR TITLE
Changed file uploader to also encrypt

### PR DESCRIPTION
**Pull request by arjunrn**
Changed the `files.yaml` schema to specify encrypted files and permissions. 

Each of the files are encrypted separately because of the limit set by KMS.

### CODE REVIEW

File: provisioner/aws.go [hunk 1]
```diff
@@ -107,6 +108,10 @@ type s3UploaderAPI interface {
 	Upload(input *s3manager.UploadInput, options ...func(*s3manager.Uploader)) (*s3manager.UploadOutput, error)
 }
 
+type kmsAPI interface {
+	Encrypt(input *kms.EncryptInput) (*kms.EncryptOutput, error)
+}
```
Comment (mikkeloscar) [inline provisioner/aws.go]: Please use the `kmsiface`: https://github.com/aws/aws-sdk-go/tree/master/service/kms/kmsiface We have not done it in other places in this project, but I think it makes more sense than defining a custom interface each time.

Here's an example of stsiface being used in a similar way: https://github.com/mikkeloscar/kube-aws-iam-controller/blob/2c39d429695d6ae624d794fa19e7755c2b9a0112/credentials_getter_test.go#L16

Comment (arjunrn) [reply provisioner/aws.go]: What magic is this :)

**Comment by aermakov-zalando** [comment]
👍

**Comment by arjunrn** [comment]
:+1:

File: provisioner/remote_files.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	defer gzw.Close()
+	tarWriter := tar.NewWriter(gzw)
+	defer tarWriter.Close()
+	for _, remoteFile := range data.Files {
+		var compressionInput []byte
+		var finalPath string
+
+		decoded, err := base64Decode(remoteFile.Data)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode data: %v for file: %s", remoteFile.Data, remoteFile.Path)
```
Comment (mikkeloscar) [inline provisioner/remote_files.go]: Maybe we should not log the data in case it's supposed to be encrypted :)

Comment (arjunrn) [reply provisioner/remote_files.go]: Fix in [PR](https://github.com/zalando-incubator/cluster-lifecycle-manager/pull/147)

**Pull request merged**